rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -921,6 +967,7 @@ void RSocketStateMachine::writeNewStream(
 }
 
 void RSocketStateMachine::writeRequestN(StreamId streamId, uint32_t n) {
+  resumeManager_->incrConsumerAllowance(streamId, n);
```

Comment:
This should be atomic with telling the resume manager about the frame. Also the two calls should be in the same place.